### DISCUSSION TITLE
component composition

### DISCUSSION THREAD

**Discussion by durgaprasadksv** [nandorojo/dripsy#205] [Q&A]
@nandorojo \
first off, that you so much for the awesome library. We have been slowly moving off styled-components. Can you please suggest on What is the best way to compose components by passing default props like we use `attrs` in styled components

```
type ButtonProps = {
  enabled: boolean,
  hasHapticFeedback: Boolean,
}

const BaseButton = styled(Button)(props => ({enabled}))
const InteractiveButton = styled(BaseButton).attrs({hasHapticFeedback: true})({})

```

**Comment by nandorojo** [comment]
hey, i'm not entirely sure with what this does...can you not just pass the prop directly? or wrap it and set a default?
